src/Core/Notifications/INotificationService.cs

```diff
@@ -45,28 +47,36 @@ public interface INotificationService : IDisposable
 internal sealed class NotificationService : INotificationService
 {
     private static readonly Guid MainWindowId = Guid.Empty;
-        
+
     private readonly IInfoBarManager infoBarManager;
     private readonly IDisabledNotificationsStorage notificationsStorage;
+    private readonly IActiveSolutionTracker activeSolutionTracker;
     private readonly IThreadHandling threadHandling;
     private readonly ILogger logger;
 
     private readonly HashSet<string> oncePerSessionNotifications = [];
 
     private AttachedNotification activeNotification;
 
+    public bool CloseOnSolutionClose { get; set; } = true;
```

Comment:
Shouldn't this be a property of the notification instead?

Reply:
Also, could rename to CloseOnSolutionChange

Reply:
Moved it to the INotification, that was true, it's much cleaner solution.

The name is fitting because we only care about solution close.